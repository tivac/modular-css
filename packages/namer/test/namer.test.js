"use strict";

const { short, long } = require("../namer");

describe("@modular-css/namers", () => {
    describe("short", () => {
        let fn;
        
        beforeEach(() => (fn = short()));
        
        it("should hash its arguments", () => {
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
        });
        
        it("should differ within files", () => {
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
            expect(fn("./fooga", ".booga")).toMatchSnapshot();
        });
        
        it("should re-use selectors for identical inputs", () => {
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
        });
    
        it("should differ between files", () => {
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
            expect(fn("./booga", ".fooga")).toMatchSnapshot();
        });
        
        it("should wrap as necessary", () => {
            for(let x = 0; x < 100; x++) {
                for(let y = 0; y < 100; y++) {
                    fn(`./fooga${x}`, `.fooga${y}`);
                }
            }
            
            expect(fn(`./fooga${15}`, `.fooga${5}`)).toMatchSnapshot();
            expect(fn(`./fooga${1}`, `.fooga${5}`)).toMatchSnapshot();
            expect(fn(`./fooga${49}`, `.fooga${33}`)).toMatchSnapshot();
            expect(fn(`./fooga${55}`, `.fooga${63}`)).toMatchSnapshot();
            expect(fn(`./fooga${26}`, `.fooga${0}`)).toMatchSnapshot();
        });
    });

    describe("long", () => {
        let fn;
        
        beforeEach(() => (fn = long("./")));
        
        it("should hash its arguments", () => {
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
        });
        
        it("should differ within files", () => {
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
            expect(fn("./fooga", ".booga")).toMatchSnapshot();
        });
        
        it("should return identical selectors for identical inputs", () => {
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
        });
    
        it("should differ between files", () => {
            expect(fn("./fooga", ".fooga")).toMatchSnapshot();
            expect(fn("./booga", ".fooga")).toMatchSnapshot();
        });
    });
});
