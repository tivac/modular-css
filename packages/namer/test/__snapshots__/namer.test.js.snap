// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modular-css/namers short should differ between files 1`] = `"AA"`;

exports[`@modular-css/namers short should differ between files 2`] = `"BA"`;

exports[`@modular-css/namers short should differ within files 1`] = `"AA"`;

exports[`@modular-css/namers short should differ within files 2`] = `"AB"`;

exports[`@modular-css/namers short should hash its arguments 1`] = `"AA"`;

exports[`@modular-css/namers short should re-use selectors for identical inputs 1`] = `"AA"`;

exports[`@modular-css/namers short should re-use selectors for identical inputs 2`] = `"AA"`;

exports[`@modular-css/namers short should wrap as necessary 1`] = `"PF"`;

exports[`@modular-css/namers short should wrap as necessary 2`] = `"BF"`;

exports[`@modular-css/namers short should wrap as necessary 3`] = `"xh"`;

exports[`@modular-css/namers short should wrap as necessary 4`] = `"zD9B"`;

exports[`@modular-css/namers short should wrap as necessary 5`] = `"aA"`;
